src/mds/MDSRank.cc

```diff
@@ -3339,6 +3339,13 @@ bool MDSRank::custom_timeout_exists(Session *s)
 void MDSRank::custom_timeout_request(Session *s, int64_t new_timeout)
 {
   sessionmap.custom_timeout_list.push_back(&s->item_custom_timeout_list);
+  stringstream ss;
+  ss << "{\"prefix\":\"fs set_session_timeout\", \"fs_name\":\""
+     << mdsmap->get_fs_name() << "\", \"session\":\"" << s->info.inst.name.num()
+     << "\", \"val\":\"" << new_timeout << "\"";
+  vector<string> command = {ss.str()};
+  bufferlist bl;
+  monc->start_mon_command(command, bl, nullptr, nullptr, nullptr);
```

Comment:
when to remove stale one

Reply:
It's labeled DNM for a reason; anything we put in for real will have to depend on the clients setting it in their mount metadata so I didn't try and cancel anything. :D